### ISSUE TITLE
关于二值化

### ISSUE THREAD

**Issue by guest88388** [gcp/leela-zero#1198]
如果leela-zero权重二值化，还能正常自对弈吗？

**Comment by anonymousAwesome** [comment]
Translation according to Google Translate:  

About binarization
"If Leela Zero has binarized weights, can it still play normally?"

Google says that binarizing an image means converting it to black and white, with no color or greyscale.  Presumably this means converting the weights from floating point integers to strict binary; ones and zeroes.  I'm gonna respond with the following text, translated through Google and verified by Bing.  

No, you need new software.
(Software translated this.)

不，你需要新的软件。
（软件翻译了这个。）

**Comment by anonymousAwesome** [comment]
Translation according to Google Translate:

About binarization
"If Leela Zero has binarized weights, can it still play normally?"

Google says that binarizing an image means converting it to black and white, with no color or greyscale. Presumably this means converting the weights from floating point integers to strict binary; ones and zeroes. I'm gonna respond with the following text, translated through Google and verified by Bing.

No, you need new software.

不，你需要新的软件。

**Comment by guest88388** [comment]
我没说要改变输入啊，只是改一下权重（不过这样就没有加速效果了）。

**Comment by Fan-Mi** [comment]
For the convenience of everyone, please speak English so that other people can join the discussion. Thanks.

**Comment by featurecat** [comment]
@guest88388 I attempted to binarize the weights file by setting values larger than the line average to 1 and 0, or 1 and -1. None of those tests were successful. So, I binarized each line according to its average. Each line contains only two different types of numbers. The net is awful, but it is slightly coherent: https://drive.google.com/open?id=1ZU-5aFcOMEEOo3M5BzlS1Uf0lZt1Ep71. I tried to make it in ternary, too: https://drive.google.com/open?id=1S9l70j3HHw4Gbflp0acNErsazCHCQsa0. It is very bad.

The most interesting network I produced was by adding 0.1 * random() * lineAverage. This produced the following net: https://drive.google.com/open?id=1Nta7kvodd9BLCCiOlj65AZUUqJCo4PIa. It appears to be completely coherent, but differs in style from official Leela Zero. It is generated randomly, so it will almost certainly be weaker than Leela Zero 122. But it may still be dan level.

I hope to maybe generate more interesting networks later. Hope these are interesting!
